### DISCUSSION TITLE
Link aggregation config not applied as a clc snippet

### DISCUSSION THREAD

**Discussion by mister2d** [flatcar/Flatcar#746] [Q&A] [answered]
Trying to bring up a DHCP `bond0` interface with no success. I based the config from this [example](https://www.flatcar.org/docs/latest/provisioning/ignition/network-configuration/#bonded-nics) in the Flatcar docs. Using this as a clc snippet doesn't appear to work. Udev has enumerated the two NICs as `eno12399` and `eno12409`.

```yaml
---
ignition:
  version: 2.0.0
networkd:
  units:
  - name: 00-eno.network
    contents: |-
      [Match]
      Name=eno12*

      [Network]
      Bond=bond0
  - name: 10-bond0.netdev
    contents: |-
      [NetDev]
      Name=bond0
      Kind=bond

      [Bond]
      Mode=802.3ad
      LACPTransmitRate=fast
      TransmitHashPolicy=layer2
      MIIMonitorSec=1s
  - name: 20-bond0.network
    contents: |-
      [Match]
      Name=bond0

      [Network]
      DHCP=true
```

**Comment by pothos** [comment, marked as answer]
This seems to be caused by ign-converter failing to handle the networkd units when transpiling v2 to v3, @tormath1 can tell more.
The workaround would be to create the units through plain file entries.

**Reply by pothos** [reply]
https://github.com/flatcar-linux/Flatcar/issues/741

**Reply by mister2d** [reply]
I was able to get it working by changing the prefix on the unit names. Thankfully this is resolved.

```yaml
---
networkd:
  units:
  - name: 60-bond0.network
    contents: |-
      [Match]
      Name=eno12*

      [Network]
      Bond=bond0
  - name: 61-bond0.netdev
    contents: |-
      [NetDev]
      Name=bond0
      Kind=bond

      [Bond]
      Mode=802.3ad
      LACPTransmitRate=slow
      TransmitHashPolicy=layer2
      MIIMonitorSec=1s
       
  - name: 62-bond0.network
    contents: |-
      [Match]
      Name=bond0

      [Network]
      DHCP=true

      [DHCPv4]
      UseDomains=true
```

**Reply by pothos** [reply]
Ok, so this was on Stable, good.
Were there any other unit files under /etc/systemd/networkd/ or /run/systemd/networkd/ that had higher priority?

**Reply by mister2d** [reply]
> Ok, so this was on Stable, good. Were there any other unit files under /etc/systemd/networkd/ or /run/systemd/networkd/ that had higher priority?

Unsure at this point but anything 10-* did not work for me. I followed the [documentation](https://www.flatcar.org/docs/latest/provisioning/ignition/network-configuration/#bonded-nics) and a few iterations on that with no success.

**Reply by tormath1** [reply]
Ok thanks for checking. Can you go back to the "failing" mode with `10-*` and check the following command:
```
ls /etc/systemd/networkd/ 
ls /run/systemd/networkd/
networkctl status eno12399
```
